src/main/java/org/shanerx/tradeshop/player/ShopUser.java

```diff
@@ -146,20 +144,23 @@ public static Sign findObservedSign(Player observer) {
      * @return Shop List containing all shops in any chunk with a block in the range that match criteria specified
      */
     public static List<Shop> findProximityShop(Location center, int range, boolean inStock, List<ItemStack> desiredCost, List<ItemStack> desiredProduct) {
-        List<ChunkSnapshot> chunksInRange = new ArrayList<>(); //Used to prevent checking a chunk more than once
+        Set<ChunkSnapshot> chunksInRange = new HashSet<>(); //Used to prevent checking a chunk more than once
         List<Shop> foundShops = new ArrayList<>();
         DataStorage dataStorage = TradeShop.getPlugin().getDataStorage();
         World world = center.getWorld();
 
-        for (int x = center.getBlockX() - range; x <= center.getBlockX() + range; x++) {
-            for (int z = center.getBlockZ() - range; z <= center.getBlockZ() + range; z++) {
-                if (world != null) {
+        if (world != null) {
+            for (int x = center.getBlockX() - range; x <= center.getBlockX() + range; x++) {
+                for (int z = center.getBlockZ() - range; z <= center.getBlockZ() + range; z++) {
                     int cX = x / 16, cZ = z / 16;
-                    if (chunksInRange.stream().noneMatch((tempChunk) -> tempChunk.getWorldName().equalsIgnoreCase(center.getWorld().getName()) && tempChunk.getX() == cX && tempChunk.getZ() == cZ)) { //If not already processed
+                    if (chunksInRange.stream().noneMatch((tempChunk) -> tempChunk.getWorldName().equals(center.getWorld().getName()) && tempChunk.getX() == cX && tempChunk.getZ() == cZ)) { //If not already processed
                         ChunkSnapshot c = world.getEmptyChunkSnapshot(cX, cZ, false, false);
                         chunksInRange.add(c); //"Mark" as processed
 
-                        foundShops.addAll(dataStorage.getMatchingShopsInChunk(c, inStock, desiredCost, desiredProduct));
+                        foundShops.addAll(dataStorage.getMatchingShopsInChunk(c, inStock, desiredCost, desiredProduct)
+                                .stream()
+                                .filter(shop -> Math.pow(shop.getShopLocation().getBlockX(), 2) + Math.pow(shop.getShopLocation().getBlockZ(), 2) < Math.pow(range, 2))
```

Comment:
What is this filtering for?

Reply:
I changed to radial search, meaning all shops within a radius R instead of. Abox RxR... this was based on the feedback received